[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59347 [TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`.**
* #59346 [TensorExpr] Handle not-specified dtypes and strides.
* #59163 [JIT] ShapeProp: add missing ops from mobilenet v3.

We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

Commit:
[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`.

We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Commit:
Update on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Chillee commented on torch/csrc/jit/tensorexpr/external_functions.cpp:
```diff
@@ -110,7 +110,10 @@ void nnc_aten_adaptive_avg_pool2d(
   int64_t H = extra_args[0];
-  int64_t W = extra_args[1];
+  int64_t W = H;
+  if (args_num > 1) {
```
In my opinion, we should always pass in all of the args, and handle this kind of default propagation at the lowerings level.

ZolotukhinM replied:
That sounds good to me, I will update the PR!

Chillee reviewed (approved):
For some of these, I think I'm pretty close to being able to codegen them.

Commit:
Update on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Commit:
Update on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Commit:
rebase on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Commit:
rebase on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Commit:
rebase on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Commit:
Update on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Commit:
Update on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

Commit:
rebase on "[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`."


We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).
